src/prefect/tasks/cubejs/cubejs_utils.py

```diff
@@ -0,0 +1,179 @@
+                deployments).
+            - security_context (str, dict): The security context to be used
+                when interacting with Cube.js APIs.
+            - secret (str): The secret string to be used, together with the
+                `security_context`, to generate the API token to pass in the
+                authorization header.
+            - wait_api_call_secs (int): Number of seconds to wait
+                between API calls.
+            - max_wait_time (int): The maximum amount of seconds to wait for
+                an API call to respond.
+        """
+        self.cube_base_url = self.get_cube_base_url(subdomain=subdomain, url=url)
+        self.api_token = self.get_api_token(
+            security_context=security_context, secret=secret
+        )
+        self.wait_api_call_secs = wait_api_call_secs
+        self.max_wait_time = max_wait_time
+
+    def get_cube_base_url(self, subdomain: str, url: str) -> str:
+        """
+        Get Cube.js base URL.
+
+        Args:
+            - subdomain (str): Cube Cloud subdomain.
+            - url (str): Cube custom URL
+                (likely to be used in Cube.js self-hosted deployments).
+
+        Returns:
+            - Cube.js API base url.
+        """
+        cube_base_url = self.__CUBEJS_CLOUD_BASE_URL
+        if subdomain:
+            cube_base_url = f"{cube_base_url.format(subdomain=subdomain)}/cubejs-api"
+        else:
+            cube_base_url = url
+        return cube_base_url
+
+    def get_query_api_url(self) -> str:
+        """
+        Get Cube.js Query API URL.
+
+        Returns:
+            - Cube.js Query API URL.
+        """
+        return f"{self.cube_base_url}/v1/load"
+
+    def get_generated_sql_api_url(self) -> str:
+        """
+        Get Cube.js Query SQL API URL.
+
+        Returns:
+            - Cube.js Query SQL API URL.
+        """
+
+        return f"{self.cube_base_url}/v1/sql"
+
+    def get_api_token(self, security_context: Union[str, Dict], secret: str) -> str:
+        """
+        Build API Token given the security context and the secret.
+
+        Args:
+            - security_context (str, dict, optional): The security context to use
+                during authentication.
+            - secret: The API secret used to generate an
+                API token for authentication.
+
+        Returns:
+            - The API Token to include in the authorization headers
+                when calling Cube.js APIs.
+        """
+        api_token = jwt.encode(payload={}, key=secret)
+        if security_context:
+
+            extended_context = security_context
+            if "exp" not in security_context and "expiresIn" not in security_context:
+                extended_context["expiresIn"] = "7d"
+            api_token = jwt.encode(
+                payload=extended_context, key=secret, algorithm="HS256"
+            )
+
+        return api_token
+
+    def get_data_from_url(self, url: str, params: Dict) -> Dict:
```

Comment:
I think you should use `self.url` instead of passing `url` in.
```suggestion
    def _get_data_from_url(self, params: Dict) -> Dict:
```